### PR TITLE
Display fake data for nonLDAP users

### PR DESCRIPTION
**Pull request by TrangEmily**
This PR is for issue #66 
Hi @armenzg, I "reverted" two diffs you showed in the issue's description. I also changed a little bit in function `findReportees` of `getAllFakeReportees.js`, because the user's email is not in fake data. This is the change:

```javascript
 const allEmails = Object.keys(completeOrg);
 const fakeEmail = (email in completeOrg) ? email : allEmails[allEmails.length - 1];

 allReportees[email] = completeOrg[fakeEmail];
 const { reportees } = completeOrg[fakeEmail];
```

However, the result looks different from the old version, so maybe I did something wrong because I don't understand the context. Please help me to check. Thank you!

This is the result of this PR

Screen Shot 2019-04-15 at 4 31 49 PM
Screen Shot 2019-04-15 at 4 31 43 PM
Screen Shot 2019-04-15 at 4 31 35 PM

And this is the result of the master branch before you merged the PR about "SSO aunthentication and fetch Taskcluster" 
Screen Shot 2019-04-15 at 3 42 01 PM
Screen Shot 2019-04-15 at 3 41 54 PM
Screen Shot 2019-04-15 at 3 41 47 PM
Screen Shot 2019-04-15 at 4 37 58 PM

### CODE REVIEW

File: src/views/Main/index.jsx [hunk 1]
```diff
@@ -73,7 +74,7 @@ class MainContainer extends Component {
 
     async getReportees(userSession, ldapEmail) {
       const secretsClient = userSession.getTaskClusterSecretsClient();
-      const partialOrg = await getAllReportees(secretsClient, ldapEmail);
+      const partialOrg = (userSession.oidcProvider === 'mozilla-auth0') ? await getAllFakeReportees(ldapEmail) : await getAllReportees(secretsClient, ldapEmail);
```
Comment (armenzg) [inline src/views/Main/index.jsx]: Call `getAllReportees` with `userSession` and make this changes within `getAllReportees`:
https://github.com/mozilla/bugzilla-dashboard/blob/master/src/utils/getAllReportees.js#L52

If you do this you will not need the files `getAllFakeReportees` and `getOrgChart`.

**Comment by TrangEmily** [comment]
Thank @armenzg I updated, however, the number still is different from the old version. Is it ok? Thanks.
Screen Shot 2019-04-16 at 8 53 05 AM
Screen Shot 2019-04-16 at 8 52 52 AM

File: src/utils/getAllReportees.js [hunk 2]
```diff
@@ -49,8 +56,16 @@ const findReportees = (completeOrg, email) => {
...
+const getAllReportees = async (userSession, ldapEmail) => {
+  let completeOrg;
+  if (userSession.oidcProvider === 'mozilla-auth0') {
+    // if non-LDAP user, get fake data
+    const people = await (await fetch('people.json')).json();
+    completeOrg = await buildOrgChartData(people);
+  } else {
+    const secretsClient = userSession.getTaskClusterSecretsClient();
+    completeOrg = await getOrgChart(secretsClient);
+  }
```
Comment (armenzg) [inline src/utils/getAllReportees.js]: Do this instead:

```javascript
  let people;
  if (userSession.oidcProvider === 'mozilla-auth0') {
    // if non-LDAP user, get fake data
    people = await (await fetch('people.json')).json();
    
  } else {
    const secretsClient = userSession.getTaskClusterSecretsClient();
    const { secret } = await await secretsClient.get(config.taskclusterSecrets.orgData);
    people = secret.employees;
  }
  completeOrg = await buildOrgChartData(people);
```

You can then get rid of the `getOrgChart` function.

**Comment by armenzg** [comment]
Please fix the message in the console:
> Warning: Failed prop type: The prop `ldapEmail` is marked as required in `Reportees`, but its value is `undefined`.

Once we have an ldapEmail set in the state we can then create a fake team for it under src/config.js

**Comment by armenzg** [comment]
I've verified that immedtiately after signing in I get this message [1]
I'm using GitHub to sign in.
I see the issue both on Firefox and Chrome.

[1]
```
Warning: Failed prop type: The prop `ldapEmail` is marked as required in `Reportees`, but its value is `undefined`.
    in Reportees (created by WithStyles(Reportees))
    in WithStyles(Reportees) (created by Route)
    in Route (at PropsRoute/index.jsx:5)
    in PropsRoute (at Main/index.jsx:258)
    in Switch (at Main/index.jsx:256)
    in Suspense (at Main/index.jsx:255)
    in div (at Main/index.jsx:245)
    in div (at Main/index.jsx:240)
    in MainContainer (created by WithStyles(MainContainer))
    in WithStyles(MainContainer) (created by Route)
    in Route (at PropsRoute/index.jsx:5)
    in PropsRoute (at App/index.jsx:98)
    in Switch (at App/index.jsx:89)
    in div (created by App)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App/index.jsx:84)
    in App (created by WithStyles(App))
    in WithStyles(App) (created by HotExportedWithStyles(App))
    in AppContainer (created by HotExportedWithStyles(App))
    in HotExportedWithStyles(App) (at src/index.jsx:5)
```

**Comment by haint504** [comment]
I updated to get the fake data for Teams tab by using TEAM_CONFIG. Please help me to guide which I should do next. Thank you.
Screen Shot 2019-04-27 at 10 39 13 PM

**Comment by TrangEmily** [comment]
I updated to get the fake data for Teams tab by using TEAM_CONFIG. Please help me to guide which I should do next. Thank you.

Screen Shot 2019-04-27 at 10 39 13 PM
P/s I deleted the previous message (I used another github account to test, and made a mistake commenting using that account).

**Comment by sylvestre** [comment]
Looks great. Armen is off for a few weeks. Let me check who can review that :)

**Comment by sylvestre** [comment]
@helfi92 could you please review this please ? :) Thanks

File: src/views/Main/index.jsx [hunk 3]
```diff
@@ -170,29 +169,36 @@ class MainContainer extends Component {
...
-          await Promise.all(Object.keys(BZ_QUERIES).map(async (metric) => {
-            const parameters = { product, component, ...BZ_QUERIES[metric].parameters };
-            team.metrics[metric] = await getBugsCountAndLink(parameters);
-          }));
-          teamComponents[teamKey] = team;
-          this.setState({ teamComponents });
-        }
-      });
+    async teamsData(userSession, partialOrg) {
+      console.log('teams data', userSession, partialOrg);
```
Comment (helfi92) [inline src/views/Main/index.jsx]: Remove `console.log`.

File: src/views/Main/index.jsx [hunk 4]
```diff
@@ -170,29 +169,36 @@ class MainContainer extends Component {
...
+      let teamComponents = {};
+      if (userSession.oidcProvider === 'mozilla-auth0') {
+        // if non-LDAP user, get fake data
+        teamComponents = TEAMS_CONFIG;
+      } else {
+        Object.entries(TEAMS_CONFIG).map(async ([teamKey, teamInfo]) => {
+          console.log('team config', teamKey, teamInfo);
```
Comment (helfi92) [inline src/views/Main/index.jsx]: Remove `console.log`.

**Comment by TrangEmily** [comment]
Thank you I removed the "console.log".

**Comment by sylvestre** [comment]
Oups, this broke the production.

Here is what I see when I look at the prod:
image

I will revert the change, sorry :(

**Comment by sylvestre** [comment]
@TrangEmily do you need help ? :)

**Comment by TrangEmily** [comment]
Hi @sylvestre, sorry I did not answer sooner. I am out of town and have limited Internet connection. Can I get back next week? Thank you.

**Comment by sylvestre** [comment]
sure, thanks!

**Pull request merged**